docs/applications/containers/beginners-guide-to-LXD-reverse-proxy/index.md

```diff
@@ -0,0 +1,668 @@
+keywords: ["container", "lxd", "lxc", "apache", "nginx", "reverse proxy", "virtual machine", "virtualization"]
+license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
+published: 2019-08-13
+modified: 2019-08-13
+modified_by:
+  name: Linode
+title: "A Beginner's Guide to LXD: Setting Up a Reverse Proxy to Host Many Websites in Containers"
+contributor:
+  name: Simos Xenitellis
+  link: https://blog.simos.info/
+external_resources:
+  - '[LXD Introduction](https://linuxcontainers.org/lxd/)'
+  - '[LXD support community](https://discuss.linuxcontainers.org/)'
+  - '[Try LXD Online](https://linuxcontainers.org/lxd/try-it/)'
+  - '[NGINX Web Server](https://docs.nginx.com/nginx/admin-guide/web-server/)'
+  - '[Apache Web Server](https://httpd.apache.org)'
+  - '[NGINX Reverse Proxy Settings](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/)'
+  - '[Proxy Protocol](https://www.haproxy.com/blog/haproxy/proxy-protocol/)'
+  - '[TLS Termination Proxy](https://en.wikipedia.org/wiki/TLS_termination_proxy)'
+---
+
+## Introduction
+
+[LXD](https://linuxcontainers.org/lxd/) (pronounced "Lex-Dee") is a system container manager build on top of Linux Containers (LXC) and is supported by [Canonical](https://canonical.com). The goal of LXD is to provide an experience similar to a virtual machine but through containerization rather than hardware virtualization. Compared to Docker for delivering applications, LXD offers nearly full operating-system functionality with additional features such as snapshots, live migrations, and storage management.
+
+In this guide you will:
+
+ - [Install and configure containers](/docs/applications/containers/beginners-guide-to-lxd-reverse-proxy/#creating-the-containers) for both NGINX and Apache web servers.
+
+ - [Learn how to install and configure a reverse proxy in a container](/docs/applications/containers/beginners-guide-to-lxd-reverse-proxy/#setting-up-the-reverse-proxy).
+
+ - [Get SSL/TLS support through Let's Encrypt certificates with automated certificate renewal](/docs/applications/containers/beginners-guide-to-lxd-reverse-proxy/#adding-support-for-https-with-let-s-encrypt).
+
+- [Troubleshooting common errors](/docs/applications/containers/beginners-guide-to-lxd-reverse-proxy/#troubleshooting).
+
+{{< note >}}
+For simplicity, the term *container* is used throughout this guide to describe the LXD system containers.
+{{< /note >}}
+
+### Before You Begin
+
+1.  Complete the [A Beginner's Guide to LXD: Setting Up an Apache Webserver In a Container](/docs/applications/containers/beginners-guide-to-lxd/) guide. At the end, the guide instructs you to create a container called `web` with the Apache web server for testing purposes. Remove this container by running the following commands.
+
+        lxc stop web
+        lxc delete web
+
+    {{< note >}}
+For this guide LXD version 3.3 or later is needed. Check the version with the following command:
```

Comment:
I added a version note about needing 3.3 with instructions about upgrading if needed as per our discussion in the comments.